### PR TITLE
Refacto getShopLogo remove hasKey usage

### PR DESCRIPTION
**Pull request by cruftex**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop 
| Description?      | Remove usage of `Configuraton::hasKey` in FrontController. This serves as a starting point for potentially remove this at other places as well
| Type?             | refacto
| Category?         | CO
| BC breaks?        | no
| Deprecations?     | no
| How to test?      | I hope that this is covered by automated tests
| Fixed ticket?     | partial, https://github.com/PrestaShop/PrestaShop/issues/30373
| Related PRs       | 
| Sponsor company   | headissue GmbH

### CODE REVIEW

**Comment by cruftex** [comment]
I just realise that it should be:

````
        $logoFileName = Configuration::get('PS_LOGO');
        if (!$logoFileName) {
            return [];
        }
````

Since the default is `false` when the configuration is not present. That should be caught by the automated tests, let's see whether the tests are running without a logo at all.

The default being `false` is actually quite unexpected. For getting an arbitrary configuration item, I would expect the default being `null` if it is not present.

**Review by Hlavtox — DISMISSED** [review]

**Review by kpodemski — DISMISSED** [review]

**Review by mflasquin — COMMENTED** [review]
hasKey is also used in getTemplateVarShop function in FrontController. Maybe you can try to change also here.

**Comment by cruftex** [comment]
@mflasquin
> hasKey is also used in getTemplateVarShop function in FrontController. Maybe you can try to change also here.

Yes, that can be changed also!

`hasKey` has two bugs: it does not load the configuration, it does not take into account the shop context. If `hasKey` would be fixed, it is basically doing almost the same as `get`.

The `FrontController` is basically the only user of `hasKey`, so, after its removal hasKey could actually be deprecated.

File: classes/controller/FrontController.php [hunk 1]
```diff
@@ -1655,6 +1655,11 @@ public function getTemplateVarShop()
         return $shop;
     }
 
+    private static function configuredImageUrl(string $configKey, $psImageUrl) {
+        $image = Configuration::get($configKey);
+        return $image ? $psImageUrl . $image : '';
+    }
```
Comment (mflasquin) [inline classes/controller/FrontController.php]:
```suggestion
    private function configuredImageUrl(string $configKey, $psImageUrl) 
    {
        $image = Configuration::get($configKey);
        
        return $image ? $psImageUrl . $image : '';
    }
```

**Review by mflasquin — DISMISSED** [review]

File: classes/controller/FrontController.php (hunk 1 above)
Comment (cruftex) [reply classes/controller/FrontController.php]: This is fixed now.

**Review by kpodemski — APPROVED** [review]

**Review by nicosomb — APPROVED** [review]

**Review by boherm — APPROVED** [review]

**Comment by hibatallahAouadni** [comment]
@PrestaShop/committers as the related issue to this PR need to be reproduced by a Developer, I will put the waiting for Dev to this PR, to be tested by a Dev 😉

**Review by sallemiines — APPROVED** [review]
Hello @cruftex 

Thanx for the PR ! LGTM QA :heavy_check_mark: 
[Tests](https://github.com/aniszr/ga.tests.ui.pr/actions/runs/5864903899) OK 

Thank you

**Comment by matks** [comment]
Thank you @cruftex

**Pull request merged**